test/attr/attr_completionhandlerasync.swift

```diff
@@ -0,0 +1,129 @@
+@completionHandlerAsync("$dollarname")
+func func3() {}
+
+// expected-error@+1:25{{argument of 'completionHandlerAsync' attribute must be an identifier or full function name}}
+@completionHandlerAsync("TypePrefix.func")
+func func4() {}
+
+// expected-error@+1{{argument of 'completionHandlerAsync' cannot be an interpolated string literal}}
+@completionHandlerAsync("interpreted \()")
+func func5() {}
+
+@completionHandlerAsync("foo" // expected-error{{expected ')' in 'completionHandlerAsync' attribute}}
+func func6() {}
+
+@completionHandlerAsync("foo", completionHandlerIndex: 2 // expected-error@:57{{expected ')' in 'completionHandlerAsync'}}
+func func7() {}
+
+// expected-error@+1{{'@completionHandlerAsync' attribute cannot be applied to this declaration}}
+@completionHandlerAsync("foo", completionHandlerIndex: 0)
+protocol SomeProto {
+  // expected-error@+1:27{{no suitable '@completionHandlerAsync' async functions 'protoFunc'}}
+  @completionHandlerAsync("protoFunc", completionHandlerIndex: 0)
+  func protoFunc(continuation: @escaping () -> Void)
+}
+
+// expected-error@+1{{'@completionHandlerAsync' attribute cannot be applied to this declaration}}
+@completionHandlerAsync("foo", completionHandlerIndex: 0)
+struct SomeStruct: SomeProto {
+  // expected-error@+1:27{{no suitable '@completionHandlerAsync' async functions 'protoFunc'}}
+  @completionHandlerAsync("protoFunc", completionHandlerIndex: 0)
+  func protoFunc(continuation: @escaping () -> Void) {}
+
+  func structFunc() async { }
+
+  @completionHandlerAsync("structFunc", completionHandlerIndex: 0)
+  func structFunc(continuation: @escaping () -> Void) { }
+
+  static func staticStructFunc() async { }
+
+  @completionHandlerAsync("staticStructFunc", completionHandlerIndex: 0)
+  static func staticStructFunc(completionHandler: @escaping () -> Void) { }
+}
+
+// expected-error@+1 {{'@completionHandlerAsync' attribute cannot be applied to this declaration}}
+@completionHandlerAsync("foo", completionHandlerIndex: 0)
+class SomeClass: SomeProto {
+  // expected-error@+1:27{{no suitable '@completionHandlerAsync' async functions 'protoFunc'}}
+  @completionHandlerAsync("protoFunc", completionHandlerIndex: 0)
+  func protoFunc(continuation: @escaping () -> Void) { }
+
+  func classFunc() async { }
+
+  @completionHandlerAsync("classFunc", completionHandlerIndex: 0)
+  func classFunc(completionHandler: @escaping () -> Void) { }
+}
+
+// ===================
+// Typechecking
+// ===================
+
+// expected-error@+2:23{{'@completionHandlerAsync' attached to async function}}
+@completionHandlerAsync("asyncFunc")
+func typecheckFunc1() async {}
+
+// expected-error@+2:6{{'@completionHandlerAsync' function 'typecheckFunc2()' must take a completion handler}}
+@completionHandlerAsync("betterFunc")
+func typecheckFunc2() {}
+
+// expected-error@+2:36{{completion handler must be a function type}}
```

Comment:
Could use the message I suggested above + a note on the completionHandler to say it's not a function type